flask_app.py

```diff
@@ -3,68 +3,81 @@
 from flask import render_template
 from flask import Flask
 
-from functions import create_html_columns, create_html_from_json_element
+grade_map = {
+    "9c": "5.15d",
+    "9b/c": "5.15c/d",
+    "9b+": "5.15c",
+    "9b/+": "5.15b/c",
+    "9b": "5.15b",
+    "9a+": "5.15a",
+    # Bouldering ---
+    "9A": "V17",
+    "8C+/9A": "V16/V17",
+    "8C+": "V16",
+    "8C/+": "V15/16",
+    "8C": "V15",
+}
 
 THIS_FOLDER = os.path.dirname(os.path.abspath(__file__))
 
 
 app = Flask("Hardest Climbs", template_folder=THIS_FOLDER)
 
-with open(os.path.join(THIS_FOLDER, 'data/lead.json'), "r") as f:
+with open(os.path.join(THIS_FOLDER, 'data/lead.json'), "r", encoding='utf-8') as f:
     lead_data = json.load(f)
 
-with open(os.path.join(THIS_FOLDER, 'data/boulder.json'), "r") as f:
+with open(os.path.join(THIS_FOLDER, 'data/boulder.json'), "r", encoding='utf-8') as f:
     boulder_data = json.load(f)
 
+def climber_ascents(climber, data):
+    # TODO More elegant solution would require better data system than JSON/dict
```

Comment:
Yeah, I've started to play around with getting the JSON data into a database that the app can use but wanna keep JSON for easy data entry.

Reply:
The amount of data is very low and it changes rarely, so maybe SQLite "cache" of JSON data would be reasonable? Any edit of JSON would result in rebuilding of SQLite database - that way editing would be as easy as it is now, but using the data would be more convenient because of SQL access.

Reply:
Yeah, I was also thinking just rebuilding the database should be easy enough given the amount of data.